estate-services/src/main/java/org/egov/ps/util/Util.java

```diff
@@ -136,4 +159,104 @@ public String getStateLevelTenantId(String tenantId) {
+	 			Recepient recepient = Recepient.builder().toUsers(toUsers).toRoles(null).build();
+	 			List<String> payTriggerList = Arrays.asList(config.getPayTriggers().split("[,]"));
+	 			Action action = null;
+	 			if(payTriggerList.contains(applicationStatus)) {
+	 				action= generateAction(applicationStatus,mobile,applicationNumber,tenantId);
+	 			}
+					events.add(Event.builder().tenantId(tenantId).description(mobileNumberToMsg.get(mobile))
+							.eventType(PSConstants.USREVENTS_EVENT_TYPE).name(PSConstants.USREVENTS_EVENT_NAME)
+							.postedBy(PSConstants.USREVENTS_EVENT_POSTEDBY).source(Source.WEBAPP).recepient(recepient)
+							.eventDetails(null).actions(action).build());
+					
+		}
+	         return events;
+		}
+		private Action generateAction(String applicationStatus,String mobile, String applicationNumber, String tenantId) {
+			   List<ActionItem> items = new ArrayList<>();
+			   String actionLink=null;
+					 actionLink = config.getPayLinkForApplication().replace("$mobile", mobile)
+					.replace("$applicationNo",applicationNumber )
+					.replace("$tenantId", tenantId);
+				actionLink = config.getUiAppHost() + actionLink;
+				ActionItem item = ActionItem.builder().actionUrl(actionLink).code(config.getPayCode()).build();
+				items.add(item);
+				return Action.builder().actionUrls(items).build();
+		}
+		
+		 /**
+	     * Fetches UUIDs of CITIZENs based on the phone number.
+	     * 
+	     * @param mobileNumbers
+	     * @param requestInfo
+	     * @param tenantId
+	     * @return
+	     */
+	    private Map<String, String> fetchUserUUIDs(Set<String> mobileNumbers, RequestInfo requestInfo, String tenantId) {
+	    	Map<String, String> mapOfPhnoAndUUIDs = new HashMap<>();
+	    	for(String mobileNo: mobileNumbers) {
+	    		try {
+	    			Object user = userDetails(requestInfo,tenantId,mobileNo);
+	    			if(null != user) {
+	    				String uuid = JsonPath.read(user, "$.user[0].uuid");
+	    				mapOfPhnoAndUUIDs.put(mobileNo, uuid);
+	    			}else {
+	        			log.error("Service returned null while fetching user for username - "+mobileNo);
+	    			}
+	    		}catch(Exception e) {
+	    			log.error("Exception while fetching user for username - "+mobileNo);
+	    			log.error("Exception trace: ",e);
+	    			continue;
+	    		}
+	    	}
+	    	return mapOfPhnoAndUUIDs;
+	    }
+	    
+	    public Object userDetails( RequestInfo requestInfo, String tenantId,String mobileNumber){
+	    	Object user=null;
+	    	StringBuilder uri = new StringBuilder();
+	    	uri.append(config.getUserHost()).append(config.getUserSearchEndpoint());
+	    	Map<String, Object> userSearchRequest = new HashMap<>();
+	    	userSearchRequest.put("RequestInfo", requestInfo);
+			userSearchRequest.put("tenantId", tenantId);
+			userSearchRequest.put("userType", "CITIZEN");
+			userSearchRequest.put("userName", mobileNumber);
+			
+			try {
+				 user = serviceRequestRepository.fetchResult(uri, userSearchRequest);
```

Comment:
These two functions should be in UserService.java. UserService already has `getUserByUUID` and `searchByUserName` which can be reused instead of creating new functions.

Reply:
Here we are fetching user details using mobile number